### PR TITLE
[OTel] gunicorn: Gunicorn Hook Support

### PR DESCRIPTION
**Pull request by CagriYonca**
Added gunicorn hook support and tested using fast api on ubuntu.

### CODE REVIEW

File: src/instana/hooks/hook_gunicorn.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    from gunicorn.workers.sync import SyncWorker
+
+    def pre_fork(config: Config, server: Arbiter, worker: SyncWorker) -> None:
+        """This is our gunicorn hook to detect and act when worker processes are forked off."""
+        logger.debug("Handling gunicorn fork...")
+        agent.handle_fork()
+
+    Config.pre_fork = pre_fork
+
+    logger.debug("Gunicorn post-fork hook applied")
```
Comment (pvital) [inline src/instana/hooks/hook_gunicorn.py:20]:
```suggestion
    logger.debug("Gunicorn pre-fork hook applied")
```

**Review by pvital — CHANGES_REQUESTED** [review]
Minor fix.

**Review by GSVarsha — APPROVED** [review]
LGTM!

**Review by pvital — APPROVED** [review]
It looks good to me.

**Pull request merged**